### PR TITLE
feat(trigger): Allow entry of explicit build number

### PR DESCRIPTION
**Pull request by alanmquach**
If for some reason the build you're after is not found or not in the list.

aquachpy_·_Pipeline_Executions

aquachpy_·_Pipeline_Executions

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
What's the scenario where Igor can't find a build to display here but doesn't blow up when we trigger a pipeline referencing that same build?

How would you feel about using a custom `menuRenderer` (we do something similar-ish in the `<AmazonImageSelectInput>`) and putting the link to select a build manually as either 
 a) the last item in the list of builds, or
 b) in a custom footer (always visible) on the menu when it's opened

**Comment by alanmquach** [comment]
The Jenkins API only returns the most recent 100. If someone needs to trigger a pipeline against an older build that still exists in Jenkins, there's no way to do it from deck.

Cool, I didn't know about custom `menuRenderer`s. Let me take a look at making it a sticky footer.

**Comment by alanmquach** [comment]
Oh, but that wouldn't work for the error or none cases.

**Comment by anotherchrisberry** [comment]
Then I approve

File: app/scripts/modules/core/src/pipeline/config/triggers/baseBuild/BaseBuildTriggerTemplate.tsx [hunk 1]
```diff
@@ -134,27 +147,38 @@ export class BaseBuildTriggerTemplate extends React.Component<
...
-        {loadError && <div className="col-md-6">Error loading builds!</div>}
-        {!buildsLoading && (
-          <div className="col-md-6">
-            {builds.length === 0 && (
-              <div>
-                <p className="form-control-static">No builds found</p>
-              </div>
-            )}
-            {builds.length > 0 && (
+        <div className="col-md-6">
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/pipeline/config/triggers/baseBuild/BaseBuildTriggerTemplate.tsx]: wow this reads surprisingly nice.  I like!

**Pull request merged**